goodtablesio/tests/plugins/github/test_tasks.py

```diff
@@ -1,17 +1,24 @@
 from unittest.mock import patch
 from goodtablesio.plugins.github import tasks
 
+# Constants
+
+OWNER = 'frictionlessdata'
+REPO = 'goodtables.io-example'
+SHA = 'd5be243487d9882d7f762e7fa04b36b900164a59'
+
 
 # Tests
 
 def test_get_job_base():
-    actual = tasks._get_job_base('frictionlessdata', 'goodtables.io-example')
-    expect = 'https://raw.githubusercontent.com/frictionlessdata/goodtables.io-example/master'  # noqa
+    actual = tasks._get_job_base(OWNER, REPO, SHA)
+    expect = 'https://raw.githubusercontent.com/frictionlessdata/'
+    expect += 'goodtables.io-example/d5be243487d9882d7f762e7fa04b36b900164a59'
     assert actual == expect
 
 
 def test_get_job_files():
-    actual = tasks._get_job_files('frictionlessdata', 'goodtables.io-example', 'd5be243487d9882d7f762e7fa04b36b900164a59')  # noqa
+    actual = tasks._get_job_files(OWNER, REPO, SHA)
```

Comment:
Well...

```________________________________________________________________________________________ test_get_job_files_fallback _________________________________________________________________________________________

_get_job_files_tree_api = <MagicMock name='_get_job_files_tree_api' id='140385352242680'>

    @patch.object(tasks, '_get_job_files_tree_api')
    def test_get_job_files_fallback(_get_job_files_tree_api):
        _get_job_files_tree_api.return_value = None
        actual = tasks._get_job_files('frictionlessdata', 'goodtables.io-example', 'd5be243487d9882d7f762e7fa04b36b900164a59')  # noqa
        expect = [
            'README.md',
            'data/invalid.csv',
            'goodtables.yml',
            'valid.csv',
        ]
>       assert actual == expect
E       assert ['LICENSE.md'..., 'valid.csv'] == ['README.md', ..., 'valid.csv']
E         At index 0 diff: 'LICENSE.md' != 'README.md'
E         Left contains more items, first extra item: 'valid.csv'
E         Use -v to get the full diff

goodtablesio/tests/plugins/github/test_tasks.py:34: AssertionError
-------------------------------------------------------------------------------------------- Captured stderr call --------------------------------------------------------------------------------------------
```

Tests should be completely isolated. When you rely on an external resource your tests are stable as long as that resource is stable. You changed some files on the remote repo and now the test is failing. Mocking actually increases the confidence that it will work in the future.